lib/ansible/modules/cloud/kubevirt/kubevirt_facts.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: kubevirt_facts
+
+short_description: Retrieve facts about a KubeVirt resources
+
+version_added: "2.5"
+
+author: KubeVirt Team (@kubevirt)
+
+description:
+  - Retrieve facts about a KubeVirt resources
+
+requirements:
+  - python >= 2.7
+  - kubevirt-python >= v0.6.0-177-gd265c3c
+'''
+
+EXAMPLES = '''
+# Gather facts about a VM called C(testvm) created on the namespace C(vms)
+- kubevirt_facts:
+    name: testvm
+    namespace: vms
+    kind: VirtualMachine
+  register: vm_facts
+- debug:
+    var: vm_facts
+
+# Gather facts for all VMIs in a cluster
+- kubevirt_facts:
+    kind: VirtualMachineInstance
+  register: all_vmis
+- debug:
+    var: all_vmis
+
+# Gather facts for all VMIRS in vms namespace
+- kubevirt_facts:
+    kind: VirtualMachineInstanceReplicaSet
+    namespace: vms
+  register: all_vms_vmirs
+- debug:
+    var: all_vms_vmirs
+
+# Gather facts for all VMIS in vms namespace with label 'app = web'
+- kubevirt_facts:
+    kind: VirtualMachineInstance
+    namespace: vms
+    label_selector
+'''
+
+RETURN = '''
+items:
+  description: "Dictionary describing the requested resource.
+                Resource attributes are mapped to dictionary
+                keys, which can be found at the following URL:
+                U(http://www.kubevirt.io/api-reference/master/definitions.html)"
+  returned: On success.
+  type: complex
+  contains:
+    api_version:
+      description: The versioned schema of this representation of an object.
+      returned: success
+      type: str
+    kind:
+      description: Represents the REST resource this object represents.
+      returned: success
+      type: str
+    metadata:
+      description: "Standard object metadata. Includes name, namespace,
+                    annotations, labels, etc."
+      returned: success
+      type: dict
+    spec:
+      description: "Specific attributes of the object. Will vary based
```

Comment:
`Will` doesn't have a subject. `Specific attributes of the object depending on the I(api_version) and I(kind).`